lumigator/python/mzai/backend/services/datasets.py

```diff
@@ -153,29 +171,51 @@ def get_dataset_s3_path(self, dataset_id: UUID) -> str:
     def delete_dataset(self, dataset_id: UUID) -> None:
         record = self._get_dataset_record(dataset_id)
 
-        # Delete from S3
-        # S3 delete is called first, since if this fails the DB delete won't take place
-        dataset_key = self._get_s3_key(record.id, record.filename)
-        self.s3_client.delete_object(Bucket=settings.S3_BUCKET, Key=dataset_key)
+        try:
+            # Delete from S3
+            # S3 delete is called first => if this fails the DB delete won't take place
+            dataset_key = self._get_s3_key(record.id, record.filename)
```

Comment:
could these details be hidden somewhere so the dataset service itself is not aware of what kind of FS is being used for this file?

Reply:
Definitely! At the moment, the DatasetService only works with S3. I think it'd be great to have an abstraction that allows us to be agnostic of the underlying storage. However, I am not sure we want to implement it from scratch (see my point re: artifact storage / experiment tracking). WDYT?